app/search.py

```diff
@@ -27,46 +53,63 @@
     "factual": resolveFactualPlace,
 }
 
-def getVenuesFromIndex(lat, lon, radius, maxNum):
-    maxNum = min(maxNum, 1000)
-    all = _getVenuesFromIndex(lat, lon, radius, "rating", maxNum)
+def getVenuesFromIndex(lat, lon, radius):
+    all = _getVenuesFromIndex(lat, lon, radius, YELP_SORT_ORDER)
     seen = set()
-    unique = (biz for biz in all if biz.id not in seen and not seen.add(biz.id))
-    rated = [biz for biz in unique if biz.rating > 3.0]
-    return rated
-
-def _getVenuesFromIndex(lat, lon, radius, sortOrder, maxNum):
-    total = 1
-    offset = 0
-    yelpVenues = []
-    while offset < total:
-        try:
-            locality = _singleSearchAPIQuery(lat, lon, offset, radius, sortOrder)
-            total = min(locality.total, maxNum)
-            yelpVenues += locality.businesses
-            offset += 20
-        except KeyboardInterrupt:
-            raise KeyboardInterrupt()
-        except Exception:
-            log.exception("Exception searching with Yelp API")
-            # give up if this is the first try – we've seen
-            # occasional failures where the location doesn't 
-            # have any existing support (e.g. Vanuatu)
-            # but also, just randon 500 errors, which we can recover from.
-            if total == 1 and offset == 0:
-                return yelpVenues
-            else:
-                offset += 20
+    unique = [biz for biz in all if biz.id not in seen and not seen.add(biz.id) and biz.location.coordinate != None]
+    log.info("Found %d unique venues with locations" % len(unique))
+    return unique
+
+def _getVenuesFromIndex(lat, lon, radius, sortOrder):
+    locality = _singleSearchAPIQuery(lat, lon, radius * YELP_RADIUS_FACTOR, sortOrder, 0)
+    log.debug("Crawling point: (%s, %s), radius: %s meters" % (lat, lon, radius))
+
+    # If the current result set is greater than the max we can iterate through,
+    # divide the search area and try again. Do this recursively so that we get
+    # the full set of places.
+    if locality.total > YELP_MAX_PER_SEARCH and radius > 50:
+        dst = sqrt(1./8) * radius
```

Comment:
Shouldn't this just be 1./sqrt(2)? The new radius should be the hypotenuse of the triangle whose legs are each r/2 (45-45-90 sides are x, x, x*sqrt(2)), so hypotenuse should be sqrt(2)*r/2, which is 2r/(2*sqrt(2)), which is r/sqrt(2).